### PR TITLE
Use go mod for dependencies management, update base image

### PR DESCRIPTION
**Pull request by vanastassiou**
Part of https://github.com/plotly/streambed/issues/14939

Test process:
- [x] Build Docker image with this code locally
- [x] Push to [my test repo on Docker Hub](https://hub.docker.com/r/vanastassiouplotly/pod-reaper/tags?page=1&ordering=last_updated)
- [x] [Update `25_configure_ha`](https://github.com/plotly/dash-deployment-server/blob/master/install/etc/my_init.d/25_configure_ha#L382) with the correct repo/image info to pull
    - This will not need to be repeated in a freeze test plan after this PR is merged
- [x] Publish release using above images for `pod-reaper` and `dash` and deploy HA cluster
 - [x] Deploy DOPSA to cluster, ensure it's running and browsable and nothing broke
- [x] In Server Manager settings, enable Dash customizations, set pod timeout to something nonstandard and short, and observe `pod-reaper` pod log activity to ensure lifetime annotations are respected (modelled after [PR implementing pod lifetime](https://github.com/plotly/on-prem/pull/82))
- [x] Delete Dash app
- [x] Push this image (after approval) to https://hub.docker.com/r/plotly/pod-reaper
    - Image tag `0.2.0` or other? @joaoalf 
- [x] Update https://github.com/plotly/on-prem/blob/master/replicated_tmpl_dash_enterprise.yaml with the new `pod-reaper` version (https://github.com/plotly/on-prem/pull/188)

### CODE REVIEW

**Comment by vanastassiou** [comment]
@joaoalf Can you review? I'm not entirely sure my test plan covers all `pod-reaper` use cases so your advice is welcome.

**Comment by joaoalf** [comment]
@vanastassiou For testing releases, if the image is declared on the `images.txt`, you can simply put your branch name as the version for the image and do the `build:yaml` and `build:release` steps as usual.

Also, on the test we want to see pod-reaper in action. You should probably set the timeout to a close value and tail the logs until you see it redeploying the app.

**Comment by joaoalf** [comment]
0dd904b appears to be empty.

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,9 @@
-FROM golang:1.13.4-alpine
+FROM golang:1.15.6-alpine
 
 RUN apk --no-cache add curl git && curl https://glide.sh/get | sh && apk del curl
```
Comment (joaoalf) [inline Dockerfile]: @vanastassiou Aren't we trying to remove glide? Do we need the `curl https://glide.sh/get | sh`?

**Review by joaoalf — COMMENTED** [review]
It looks good in general. It needs more tests and a couple of clean-ups.

File: Dockerfile (hunk 1 above)
Comment (vanastassiou) [reply Dockerfile]: You're right, this was a mistake.

**Comment by vanastassiou** [comment]
@joaoalf OK, finally found my way around all the blockers. The `pod-reaper` logs are showing the activity I expect, so this looks good to me. Please re-review.

**Review by joaoalf — CHANGES_REQUESTED** [review]
> Update https://github.com/plotly/on-prem/blob/master/replicated_tmpl_dash_enterprise.yaml with the new pod-reaper version

@vanastassiou do you mind adding this to images.txt instead?

> pod-reaper pod log activity to ensure lifetime annotations are respected

I think we should check the pod lifetime after the pod-reaper run to ensure it's a new pod.

I imagine that we forked this repository because it appears to be unmaintained but, I just discovered this other one (https://github.com/target/pod-reaper) that show recent activity and they in fact recently did the same work as you (https://github.com/target/pod-reaper/pull/58). Do you mind creating an issue to evaluate the possibility to give up our fork in favor of their (or another maintained) repository?

I will 💃🏼 as soon as you do the `images.txt` thing.

**Review by joaoalf — APPROVED** [review]
In fact, the image.txt part is not on this repos so... 💃🏼 if you're satisfied with the tests.

**Pull request merged**